sdk/src/main/scala/com/horizen/account/state/StateDbAccountStateView.scala

```diff
@@ -179,18 +179,20 @@ class StateDbAccountStateView(stateDb: StateDB, messageProcessors: Seq[MessagePr
       txIndex: Int,
       blockGasPool: GasPool,
       blockContext: BlockContext,
-      finalizeChanges: Boolean = true
+      finalizeChanges: Boolean = true,
+      signatureRequired: Boolean = true
   ): Try[EthereumConsensusDataReceipt] = Try {
     if (!tx.isInstanceOf[EthereumTransaction])
       throw new IllegalArgumentException(s"Unsupported transaction type ${tx.getClass.getName}")
 
     val ethTx = tx.asInstanceOf[EthereumTransaction]
 
-    // should never happen if the tx has been accepted in mempool.
+    // The signature check is skipped for some specific cases (condition checked with signatureRequired flag)
+    // but it should never happen if the tx has been accepted in mempool.
     // In some negative test scenario this can happen when forcing an unsigned tx to be forged in a block.
     // In this case the 'from' attribute in the msg would not be 
     // set, and it would be difficult to rootcause the reason why gas and nonce checks would fail
-    if (!ethTx.isSigned)
+    if (signatureRequired && !ethTx.isSigned)
```

Comment:
We don't need this with a bit of refactoring, see other comments.

Reply:
`signatureRequired` flag removed from method signature and logic